### PR TITLE
Add netcoreapp1.0 target to Nancy.Authentication.Basic.Tests

### PR DESCRIPTION
**Pull request by blairconrad**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/NancyFx/Nancy/pulls) open
- [x] I have verified that I am following the Nancy [code style guidelines](https://github.com/NancyFx/Nancy/blob/45238076ad0b7f6ecabd6bae8469e30458d02efe/CONTRIBUTING.md#style-guidelines)
- [x] I have provided test coverage for my change (where applicable)

### Description

Adds netcoreapp1.0 target to  Nancy.Authentication.Basic.Tests, and runs it on Windows and Unix

Supports #2612.

### CODE REVIEW

File: test/Nancy.Authentication.Basic.Tests/project.json [hunk 1]
```diff
@@ -17,13 +18,22 @@
...
+            "buildOptions": {
+                "define": [ "CORE" ]
+            },
+            "dependencies": {
+                "Microsoft.NETCore.App": {
+                    "type": "platform",
+                    "version": "1.0.0"
+                }
+            },
+            "imports": [ "netstandard1.6" ]
```
Comment (jchannon) [inline test/Nancy.Authentication.Basic.Tests/project.json]: Shouldn't need that import

Comment (blairconrad) [inline test/Nancy.Authentication.Basic.Tests/project.json]: Huh. So I don't! I am sure I had to add it or something like it back over at FakeItEasy. Removed. Thanks. Also, we don't use the `"CORE"` definition, so I struck that while I'm at it.

**Comment by blairconrad** [comment]
As you can see, I updated FIE to 3.0.0. And I think I've made all the other changes that were requested.
The Travis build failed, but that was due to our old friend `StaticConfiguration`. Do we just repush the build?

Also, I'm lining up more changes of this nature (and I can batch them up more, if that helps), but am loathe to push until this PR is merged.

**Comment by thecodejunkie** [comment]
I've just kicked of the travis build again. We're looking at closing down a bunch of pull request and then migrate away from project.json, to the new and supported project format (+ get rid of our .MSBuild hack)

File: test/Nancy.Tests/project.json [hunk 2]
```diff
@@ -2,8 +2,8 @@
     "version": "1.4.1.0",
     "dependencies": {
         "dotnet-test-xunit": "2.2.0-preview2-build1029",
-        "FakeItEasy": "3.0.0-rc001",
-        "FakeItEasy.Analyzer": "3.0.0-rc001",
+        "FakeItEasy": "3.0.0",
+        "FakeItEasy.Analyzer": "3.0.0",
```
Comment (thecodejunkie) [inline test/Nancy.Tests/project.json]: Just out of curiosity, what does the analyzer do?

Comment (blairconrad) [inline test/Nancy.Tests/project.json]: It tells you when you've an unused call specification, like `A.CallTo(()=>fake.Method());` with no instructions for the fake, and it also warns if you try to configure a method that is not virtual (the latter would be caught at runtime as well, but why not point out misuse as early as possible?).

See http://fakeiteasy.readthedocs.io/en/stable/analyzer/ for more.

Hmm. On closer inspection, it says nearly exactly that, so maybe save yourself a click.

**Comment by blairconrad** [comment]
> I've just kicked of the travis build again. We're looking at closing down a bunch of pull request and then migrate away from project.json, to the new and supported project format (+ get rid of our .MSBuild hack)

Does this mean that, even if this PR gets merged, I should hold off converting more tests projects until the new project format is in place?

**Comment by thecodejunkie** [comment]
I think there's enough room/time to get a couple more in before we've closed of the 5-10 PRs that we want in before migrating :P

**Comment by blairconrad** [comment]
Challenge accepted.

File: test/Nancy.Tests/project.json (hunk 2 above)
Comment (blairconrad) [inline test/Nancy.Tests/project.json]: Oh, @thecodejunkie, here's a fun example that comes to mind for no particular reason.

```
Nancy/test/Nancy.ViewEngines.Razor.Tests/TextResourceFinderFixture.cs(46,46): warning FakeItEasy0002: Non virtual member '_' cannot be intercepted.
```

**Comment by blairconrad** [comment]
Thanks for the merge, @thecodejunkie!

**Pull request merged**